doc/internal/man3/ossl_namemap_new.pod

```diff
@@ -3,7 +3,7 @@
 =head1 NAME
 
 ossl_namemap_new, ossl_namemap_free, ossl_namemap_stored,
-ossl_namemap_add, ossl_namemap_name, ossl_namemap_number
+ossl_namemap_add, ossl_namemap_num2names, ossl_namemap_name2num
```

Comment:
Just wondering what the use case for `ossl_namemap_num2names` is?

I'm having difficulty thinking of a reasonable one.  If this function is omitted, the code can be significantly simplified.

The best I can come up with is to print the names correspond to an ID but that shouldn't be time critical and iterating over the hash table is probably acceptable.

Reply:
I can imagine that down the line, it could help applications print the set of names for an algorithm. I believe we have a lister of ciphers that does this with the help of the object db... Of course, that will require a bit of wrapping as these functions aren't directly available to the application, but still...

Reply:
"I can imagine that down the line, it could help"

I strongly encourage that this release only include what is needed.  You can always add things later, once you have a better understanding of the use-case and requirements. You cannot remove things, and modifying things if it's not quite right is difficult.

Reply:
This is an *internal* API, so removing what turns out to be unused is very easy.

Reply:
The point was that a less complex implementation is possible if this is missing -- removing an internal API is easy, remember to rewrite the code into the simpler form isn't.

For printing, the slow path of iterating through all elements of the name to int hash table seems reasonable enough.